### PR TITLE
Add iterator over all query results.

### PR DESCRIPTION
**Pull request by cezarfx**
Added new method query_iterable(QueryRequest) -> QueryIterableResult in NoSQLHandle class.
The result of it is an iterable, ie. it can provide iterators that iterate over all results of a query.

### CODE REVIEW

File: src/borneo/operations.py [hunk 1]
```diff
@@ -2230,6 +2230,37 @@ def copy_internal(self):
         internal_req.is_internal = True
         return internal_req
 
+    def copy(self):
+        """
+        Copies the query request to start query results from the beginning.
+
+        :versionadded: 5.3.5
```
Comment (gmfeinberg) [inline src/borneo/operations.py:2237]: versionadded needs to be updated

File: src/borneo/operations.py [hunk 2]
```diff
@@ -3583,7 +3615,7 @@ def get_retry_stats(self):
         return self._retry_stats
 
     def get_write_units(self):
-        # Internal use only.
+        # Internal use only. ??? shouldn't be write_units ?
```
Comment (gmfeinberg) [inline src/borneo/operations.py:3618]: remove stray comment here

File: src/borneo/operations.py [hunk 3]
```diff
@@ -4469,6 +4506,191 @@ def _compute(self):
...
+
+    def __str__(self):
+        # type: () -> str
+        res = 'QueryIterableResult(' + str(self.request) + ')'
+        return res + '\n'
+
+    def get_read_kb(self):
+        # type: () -> int
+        """
+        Returns the read throughput consumed by this operation, in KBytes. This
```
Comment (gmfeinberg) [inline src/borneo/operations.py:4563]: should mention that this (and write_kb) are cumulative, based on the units consumed by all remote calls generated by the iterator

File: src/borneo/operations.py [hunk 4]
```diff
@@ -5132,6 +5354,35 @@ def add_exception(self, e):
...
         self._exception_map[e] = num
 
+    def get_exceptions_map(self):
+        # type: () -> dict[Object, int]
+        """
+        Internal use only.
+
+        Returns the map of exceptions.
+
+        :versionadded: 5.3.5
```
Comment (gmfeinberg) [inline src/borneo/operations.py:5364]: update to next release (please find all of these)

File: src/borneo/operations.py (hunk 1 above)
Comment (cezarfx) [reply src/borneo/operations.py]: Increased to 3.5.6

File: src/borneo/operations.py (hunk 2 above)
Comment (cezarfx) [reply src/borneo/operations.py:3618]: removed

File: src/borneo/operations.py (hunk 3 above)
Comment (cezarfx) [reply src/borneo/operations.py:4563]: done

File: src/borneo/operations.py (hunk 4 above)
Comment (cezarfx) [reply src/borneo/operations.py]: done

**Pull request merged**